--- /smaug/src/act_wiz.c	Thu Feb  6 09:22:47 1997
+++ /game/dist/src/act_wiz.c	Fri May 30 17:02:44 1997
@@ -1251,7 +1251,7 @@
 	get_curr_con(victim),
 	get_curr_cha(victim),
 	get_curr_lck(victim) );
-    if (IS_VAMPIRE(victim) && !IS_NPC(victim))
+    if ((IS_VAMPIRE(victim) && !IS_NPC(victim)) || (IS_DEMON(victim) && !IS_NPC(victim)))
     ch_printf( ch, "Hps: %d/%d   Blood: %d/%d   Move: %d/%d   Practices: %d\n\r",
         victim->hit,         victim->max_hit,  
         victim->pcdata->condition[COND_BLOODTHIRST], 10 + victim->level,
