--- /smaug/src/tables.c	Fri Mar  7 15:53:32 1997
+++ /game/dist/src/tables.c	Fri May 30 17:31:54 1997
@@ -73,6 +73,7 @@
     if ( !str_cmp( name, "spell_dispel_magic" ))     return spell_dispel_magic;
     if ( !str_cmp( name, "spell_dream" ))	     return spell_dream;
     if ( !str_cmp( name, "spell_earthquake" ))	     return spell_earthquake;
+    if ( !str_cmp( name, "spell_enchant_armor" ))    return spell_enchant_armor;
     if ( !str_cmp( name, "spell_enchant_weapon" ))   return spell_enchant_weapon;
     if ( !str_cmp( name, "spell_energy_drain" ))     return spell_energy_drain;
     if ( !str_cmp( name, "spell_faerie_fire" ))	     return spell_faerie_fire;
@@ -123,6 +125,9 @@
     if ( !str_cmp( name, "spell_hand_of_chaos" ))    return spell_hand_of_chaos;
     if ( !str_cmp( name, "spell_disruption" ))	     return spell_disruption;
     if ( !str_cmp( name, "spell_sonic_resonance" ))  return spell_sonic_resonance;
+    if ( !str_cmp( name, "spell_death" ))            return spell_death;
+    if ( !str_cmp( name, "spell_assassinate" ))      return spell_assassinate;
     if ( !str_cmp( name, "spell_mind_wrack" ))	     return spell_mind_wrack;
     if ( !str_cmp( name, "spell_mind_wrench" ))	     return spell_mind_wrench;
     if ( !str_cmp( name, "spell_revive" ))	     return spell_revive;
@@ -132,6 +137,7 @@
     if ( !str_cmp( name, "spell_galvanic_whip" ))    return spell_galvanic_whip;
     if ( !str_cmp( name, "spell_magnetic_thrust" ))  return spell_magnetic_thrust;
     if ( !str_cmp( name, "spell_quantum_spike" ))    return spell_quantum_spike;
+    if ( !str_cmp( name, "spell_grasp_suspiria" ))   return spell_grasp_suspiria;
     if ( !str_cmp( name, "spell_black_hand" ))	     return spell_black_hand;
     if ( !str_cmp( name, "spell_black_fist" ))	     return spell_black_fist;
     if ( !str_cmp( name, "spell_black_lightning" ))  return spell_black_lightning;
@@ -604,6 +610,7 @@
     if ( spell == spell_dispel_magic )	    return "spell_dispel_magic";
     if ( spell == spell_dream )		    return "spell_dream";
     if ( spell == spell_earthquake )	    return "spell_earthquake";
+    if ( spell == spell_enchant_armor )     return "spell_enchant_armor";
     if ( spell == spell_enchant_weapon )    return "spell_enchant_weapon";
     if ( spell == spell_energy_drain )	    return "spell_energy_drain";
     if ( spell == spell_faerie_fire )	    return "spell_faerie_fire";
@@ -663,6 +672,7 @@
     if ( spell == spell_galvanic_whip )	    return "spell_galvanic_whip";
     if ( spell == spell_magnetic_thrust )   return "spell_magnetic_thrust";
     if ( spell == spell_quantum_spike )	    return "spell_quantum_spike";
+    if ( spell == spell_grasp_suspiria )    return "spell_grasp_suspiria";
     if ( spell == spell_black_hand )	    return "spell_black_hand";
     if ( spell == spell_black_fist )	    return "spell_black_fist";
     if ( spell == spell_black_lightning )   return "spell_black_lightning";
